api/envoy/extensions/filters/udp/dns_filter/v3alpha/dns_filter.proto

```diff
@@ -44,15 +46,27 @@ message DnsFilterConfig {
   // in a client context. This message will contain the timeouts, retry,
   // and forwarding configuration for Envoy to make DNS requests to other
   // resolvers
+  //
+  // [#next-free-field: 6]
   message ClientContextConfig {
     // Sets the maximum time we will wait for the upstream query to complete
     // We allow 5s for the upstream resolution to complete, so the minimum
     // value here is 1. Note that the total latency for a failed query is the
     // number of retries multiplied by the resolver_timeout.
     google.protobuf.Duration resolver_timeout = 1 [(validate.rules).duration = {gte {seconds: 1}}];
 
+    // This field was used for `dns_resolution_config` in Envoy 1.19.0 and
+    // 1.19.1.
+    // Control planes should set this field for Envoy 1.19.0 and 1.19.1
+    // clients should fork the protobufs and change this field to type
+    // `DnsResolutionConfig`.
+    // Control planes that need to simultaneously support Envoy 1.18 should
+    // avoid Envoy 1.19.0 and 1.19.1.
```

Comment:
Add `// [#not-implemented-hide:]` here if this PR will contain only the API changes?

Or we could probably bring back the [code](https://github.com/envoyproxy/envoy/blob/release/v1.18/source/extensions/filters/udp/dns_filter/dns_filter.cc#L143-L148) to support this deprecated config (`upstream_resolvers`) on this PR. (Preferable since all the changes will be intact)

Reply:
good idea, fixing.